dnf-behave-tests/dnf/steps/cmd.py

```diff
@@ -315,6 +316,20 @@ def then_stderr_contains_lines_matching(context):
             raise AssertionError("Stderr doesn't contain line matching: %s" % line)
 
 
+@behave.then("dnf5 stdout matches line by line")
+def then_dnf5_stdout_matches(context):
+    if context.dnf5daemon_mode:
+        return
```

Comment:
To be honest I was happy we got rid of these steps that do nothing when the mode doesn't match.
I would like something like this more:
```gherkin
And stdout matches line by line
    """
    Removed 1 files, 1 directories \(total of [0-9]* KiB\). 0 errors occurred.
    """
    or in dnf5daemon mode
    """ 
    Cache successfully cleaned.
    """
```
However that doesn't seem to be possible.

The closest thing seems to be:
```gherkin
And stdout matches line by line depending on mode
    """
    Removed 1 files, 1 directories \(total of [0-9]* KiB\). 0 errors occurred.
    ---
    Cache successfully cleaned.
    """
```
Which is not great.

Reply:
Perhaps for single line outputs we could do:
```gherkin
And stdout matches depending on mode
    | mode        | line                                                                       |
    | default     | Removed 1 files, 1 directories \(total of [0-9]* KiB\). 0 errors occurred. |
    | dnf5daemon  | Cache successfully cleaned.                                                |
```
But I suppose I could file a separate issue for this if you prefer since we should focus on fixing the CI now.

Reply:
The second example (stdout matches line by line depending on mode) would be doable. The third I don't like very much because of limited usability.

I could also return to my original approach, which was to test for:
`(Removed 1 files, 1 directories \(total of [0-9]* KiB\). 0 errors occurred.|Cache successfully cleaned.)`
Which is similar to the "stdout matches line by line depending on mode" except for not defining which mode should match which. But the probability of a bug that would completely change the message and wouldn't be intentional seems quite low to me.

Reply:
> I could also return to my original approach, which was to test for:
> `(Removed 1 files, 1 directories \(total of [0-9]* KiB\). 0 errors occurred.|Cache successfully cleaned.)`
> Which is similar to the "stdout matches line by line depending on mode" except for not defining which mode should match which. But the probability of a bug that would completely change the message and wouldn't be intentional seems quite low to me.

Agreed, for me it would be the preferred solution.

Reply:
Ok, I changed it.